core/controllers/editor.py

```diff
@@ -71,6 +71,45 @@ def _require_valid_version(version_from_payload, exploration_version):
 }
 
 
+def replace_ids_with_names_in_dict(exploration_id):
+    """Replace ids with the names and return rights as a dictionary.
+
+    Args:
+        exploration_id: str. ID of the exploration.
+
+    Returns:
+        dict. The dictionary where ids are replaced with corresponding names.
+    """
+
+    rights_dict = rights_manager.get_exploration_rights(
+        exploration_id).to_dict()
+
+    if rights_dict['community_owned']:
+        rights_dict['owner_names'] = []
+        rights_dict['editor_names'] = []
+        rights_dict['voice_artist_names'] = []
+        rights_dict['viewer_names'] = []
+    else:
+        rights_dict['owner_names'] = (
+            user_services.get_human_readable_user_ids(
+                rights_dict['owner_ids']))
+        rights_dict['editor_names'] = (
+            user_services.get_human_readable_user_ids(
+                rights_dict['editor_ids']))
+        rights_dict['voice_artist_names'] = (
+            user_services.get_human_readable_user_ids(
+                rights_dict['voice_artist_ids']))
+        rights_dict['viewer_names'] = (
+            user_services.get_human_readable_user_ids(
+                rights_dict['viewer_ids']))
```

Comment:
As we are have almost same 4 lines, can we use for in this function?

Reply:
Yeah, I think we can use for. I'll do that.